app/lib/tufts/workflow_status.rb

```diff
@@ -0,0 +1,41 @@
+module Tufts
+  module WorkflowStatus
+    def self.status(id)
+      work_id = id
+      solr_doc_status = solr_status(work_id)
+      if solr_doc_status == 'unpublished' || solr_doc_status == 'deposited'
+        'unpublished'
+      else
+        # Only look for this status in Fedora if necessary
+        published_or_edited(id)
+      end
+    end
+
+    def self.solr_status(id)
+      document = SolrDocument.find(id)
+      begin
+        document['workflow_state_name_ssim'][0]
+      rescue
+        'unpublished'
+      end
+    end
+
+    def self.draft_exist?(id)
+      Rails.root.join('tmp', 'drafts', id).exist?
```

Comment:
this path is [configurable](https://github.com/curationexperts/epigaea/blob/60241628c6b4d7ab1185f9943e499f95f166d57d/app/lib/tufts/draft.rb#L27), and draft storage in general is otherwise hidden behind the `Draft` interface. I'd prefer not to reach into that logic. Instead, let's use `Draft#exists?`

Reply:
I don't think that's a public method, should I include Tufts::Draftable?

Reply:
It's definitely a public method. https://github.com/curationexperts/epigaea/blob/master/app/lib/tufts/draft.rb#L91-L95

I think your options are:

`GenericObject.new(id: id).draft_saved?`

or

```ruby
work = GenericObject.new(id: id)
Tufts::Draft.from_model(work).exists?
```

Reply:
Whenever I use those, I get this error
 `Attempting to recreate existing ldp_source: `http://127.0.0.1:8984/rest/dev/3x/81/6m/60/3x816m604'
`

Reply:
What about.. 

```
      Rails.application.config.drafts_storage_dir.join(id).exist?

```

Reply:
@little9 you shouldn't be hitting anything `LDP` related with `new`. Are you sure you aren't doing `create` or `find`?

Reply:
Sorry, my mistake. `ActiveFedora` is extremely unforgiving here. I'm fine with this as is for now.

Reply:
I opened an issue, because this is quite bad. https://github.com/samvera/active_fedora/issues/1285